R/tidybulk_utilities.R

```diff
@@ -0,0 +1,380 @@
+#' This is a generalisation of ifelse that accepts an object and return an objects
+#'
+#' @keywords internal
+#' @noRd
+#'
+#' @import dplyr
+#' @import tidyr
+#' @importFrom purrr as_mapper
+#' @importFrom magrittr equals
+#'
+#' @param .x A tibble
+#' @param .p A boolean
+#' @param .f1 A function
+#' @param .f2 A function
+#'
+#' @return A tibble
+ifelse_pipe = function(.x, .p, .f1, .f2 = NULL) {
+  switch(.p %>% not() %>% sum(1),
+         as_mapper(.f1)(.x),
+         if (.f2 %>% is.null %>% not())
+           as_mapper(.f2)(.x)
+         else
+           .x)
+}
+
+#' as_SummarizedExperiment
+#'
+#' @keywords internal
+#' @noRd
+#' 
+#' @description as_SummarizedExperiment creates a `SummarizedExperiment` object from a `tbl` or `tidybulk` tbl formatted as | <SAMPLE> | <TRANSCRIPT> | <COUNT> | <...> |
+#'
+#' @importFrom rlang enquo
+#' @importFrom rlang quo_name
+#' @importFrom rlang quo_squash
+#' @importFrom utils data
+#' @importFrom tidyr pivot_longer
+#'
+#' @param .data A tibble
+#' @param .sample The name of the sample column
+#' @param .transcript The name of the transcript/gene column
+#' @param .abundance The name of the transcript/gene abundance column
+#'
+#' @return A `SummarizedExperiment` object
+as_SummarizedExperiment = function(.data,
+                                    .sample = NULL,
+                                    .transcript = NULL,
+                                    .abundance = NULL) {
+  
+  # Get column names
+  .sample = enquo(.sample)
+  .transcript = enquo(.transcript)
+  .abundance = enquo(.abundance)
+  col_names = get_sample_transcript_counts(.data, .sample, .transcript, .abundance)
+  .sample = col_names$.sample
+  .transcript = col_names$.transcript
+  .abundance = col_names$.abundance
+  
+  # Check if package is installed, otherwise install
+  if (find.package("SummarizedExperiment", quiet = TRUE) %>% length %>% equals(0)) {
+    message("Installing SummarizedExperiment")
+    if (!requireNamespace("BiocManager", quietly = TRUE))
+      install.packages("BiocManager", repos = "https://cloud.r-project.org")
+    BiocManager::install("SummarizedExperiment", ask = FALSE)
+  }
+  if (find.package("S4Vectors", quiet = TRUE) %>% length %>% equals(0)) {
+    message("Installing S4Vectors")
+    if (!requireNamespace("BiocManager", quietly = TRUE))
+      install.packages("BiocManager", repos = "https://cloud.r-project.org")
+    BiocManager::install("S4Vectors", ask = FALSE)
+  }
```

Comment:
These two dependencies can be added to `Imports` in DESCRIPTION, and this code deleted

Reply:
This will also solve one WARNING in the tests

Reply:
These lines have been removed from the most recent commit. SummarizedExperiment and S4Vectors were already present in the Imports listl